src/numerics/IDAIntegrator.cpp

```diff
@@ -0,0 +1,588 @@
+                               "IDASetId failed.");
+        }
+    }
+
+    flag = IDASetErrHandlerFn(m_ida_mem, &ida_err, this);
+    if (flag != IDA_SUCCESS) {
+        throw CanteraError("IDAIntegrator::initialize",
+                            "IDASetErrHandlerFn failed.");
+    }
+
+    if (m_itol == IDA_SV) {
+        flag = IDASVtolerances(m_ida_mem, m_reltol, m_abstol);
+    } else {
+        flag = IDASStolerances(m_ida_mem, m_reltol, m_abstols);
+    }
+    if (flag != IDA_SUCCESS) {
+        if (flag == IDA_MEM_FAIL) {
+            throw CanteraError("IDAIntegrator::initialize",
+                               "Memory allocation failed.");
+        } else if (flag == IDA_ILL_INPUT) {
+            throw CanteraError("IDAIntegrator::initialize",
+                               "Illegal value for IDAInit input argument.");
+        } else {
+            throw CanteraError("IDAIntegrator::initialize",
+                               "IDAInit failed.");
+        }
+    }
+
+    flag = IDASetUserData(m_ida_mem, &func);
+    if (flag != IDA_SUCCESS) {
+        throw CanteraError("IDAIntegrator::initialize",
+                           "IDASetUserData failed.");
+    }
+    applyOptions();
+}
+
+void IDAIntegrator::reinitialize(double t0, FuncEval& func)
+{
+    m_t0 = t0;
+    m_time = t0;
+    func.getState(NV_DATA_S(m_y));
+    m_func = &func;
+    func.clearErrors();
+
+    int result = IDAReInit(m_ida_mem, m_t0, m_y, m_ydot);
+    if (result != IDA_SUCCESS) {
+        throw CanteraError("IDAIntegrator::reinitialize",
+                           "IDAReInit failed. result = {}", result);
+    }
+    applyOptions();
+}
+
+void IDAIntegrator::applyOptions()
+{
+    if (m_type == DENSE + NOJAC) {
+        sd_size_t N = static_cast<sd_size_t>(m_neq);
+        #if CT_SUNDIALS_VERSION >= 30
+            SUNLinSolFree((SUNLinearSolver) m_linsol);
+            SUNMatDestroy((SUNMatrix) m_linsol_matrix);
+            m_linsol_matrix = SUNDenseMatrix(N, N);
+            #if CT_SUNDIALS_USE_LAPACK
+                m_linsol = SUNLapackDense(m_y, (SUNMatrix) m_linsol_matrix);
+            #else
+                m_linsol = SUNDenseLinearSolver(m_y, (SUNMatrix) m_linsol_matrix);
+            #endif
+            IDADlsSetLinearSolver(m_ida_mem, (SUNLinearSolver) m_linsol,
+                                  (SUNMatrix) m_linsol_matrix);
+        #else
+            #if CT_SUNDIALS_USE_LAPACK
+                IDALapackDense(m_ida_mem, N);
+            #else
+                IDADense(m_ida_mem, N);
+            #endif
+        #endif
+    } else if (m_type == DIAG) {
+        throw CanteraError("IDAIntegrator::applyOptions",
+                           "Cannot use a diagonal matrix with IDA.");
+    } else if (m_type == GMRES) {
+        #if CT_SUNDIALS_VERSION >= 30
+            m_linsol = SUNSPGMR(m_y, PREC_NONE, 0);
+            IDASpilsSetLinearSolver(m_ida_mem, (SUNLinearSolver) m_linsol);
+        #else
+            IDASpgmr(m_ida_mem, PREC_NONE, 0);
```

Comment:
```suggestion
            IDASpgmr(m_ida_mem, 0);
```
Should fix the compilation error with Sundials 2.x seen in Travis.